python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -49,26 +53,17 @@ def __init__(
         self,
         dtype: plc.DataType,
         sub_expr: Expr,
-        *children: Expr,
+        *children: FusedExpr,
     ):
         self.dtype = dtype
         self.sub_expr = sub_expr
+        assert all(isinstance(c, FusedExpr) for c in children)
         self.children = children
         self.is_pointwise = sub_expr.is_pointwise
-
-    def do_evaluate(
-        self,
-        df: DataFrame,
-        *,
-        context: ExecutionContext = ExecutionContext.FRAME,
-        mapping: Mapping[Expr, Column] | None = None,
-    ) -> Column:  # pragma: no cover
-        """Evaluate this expression given a dataframe for context."""
-        return self.sub_expr.evaluate(df, context=context, mapping=mapping)
-
-    def collect_agg(self, *, depth: int) -> AggInfo:  # pragma: no cover
-        """Collect information about aggregations in groupbys."""
-        return self.sub_expr.collect_agg(depth=depth)
+        assert all(
```

Comment:
Maybe a pre-mature optimization, but worth checking that we aren't making a ton of these `FusedExpr`s, such that this assertion will have an impact on the runtime for large graphs.